src/rgw/rgw_sal_motr.cc

```diff
@@ -1307,13 +1307,24 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
         auto iter = vals[i].cbegin();
         ent.decode(iter);
         if (params.list_versions || ent.is_visible()) {
-          if (ent.key.name == params.marker.name &&
+          if (ent.key.name == params.marker.name) {
               // filter out versions which go before marker.instance
-              ((!null_ent.key.empty() && params.marker.instance == "null" &&
+              // null_ent is not yet populated
+              if (null_ent.key.name == "") {
+                 if ((params.marker.instance == "" && 
+                      ent.key.instance == "") || 
+                      (ent.key.instance != "" && 
+                      ent.key.instance < params.marker.instance))
```

Comment:
I see some handling was done previously for `params.marker.instance == "null"` and that condition is not checked now. 
so testing of `list-object-versions` api is also necessary if not done already.

Reply:
Hi @sumedhak27 when we check the absence value for `params.marker.instance` is `""`.
For `params.marker.instance == "null"` condition was not satisfying

Reply:
Since this API logic is bit tricky and the changes overlaps with PR #349, I would suggest to first merge PR #349 and rebase this PR and create a custom build to test all the possible cases with list api before merging into main.

Reply:
Done. Rebased with main (#349) changes

Reply:
> I see some handling was done previously for params.marker.instance == "null" and that condition is not checked now.
so testing of list-object-versions api is also necessary if not done already.

Is this tested now @vastradparayya? I only have concern about this `null` version handling now.